spark_auto_mapper/automappers/automapper.py

```diff
@@ -41,11 +43,21 @@ def transform_with_data_frame(
         return df
 
     def transform(self, df: DataFrame) -> DataFrame:
-        assert self.keys and len(self.keys) > 0
         # if source_view is specified then load that else assume that df is the source view
         source_df: DataFrame = df.sql_ctx.table(
             self.source_view
         ) if self.source_view else df
+        # if keys are not specified then add a __row_id column to both the source and the destination
+        #   and use that as key
+        if not self.keys or len(self.keys) == 0:
+            assert not self.view or not SparkHelpers.spark_table_exists(
```

Comment:
Is this assert required? I can think of a case where we want to rename columns in a view, so `self.view` and `self.source_view` would refer to the same view and `self.keys` would be `None`. Is that type of view manipulation discouraged?

Reply:
If we create new ids and the source table already has ids then they will not match.  If you’re adding yo an existing table in memory then you should have keys in both tables